sonarqube/datadog_checks/sonarqube/check.py

```diff
@@ -16,193 +16,88 @@ class SonarqubeCheck(AgentCheck):
-                default=True,
-            )
-            exclude_metric = self.create_metric_matcher(
-                self.compile_metric_patterns(config, 'exclude') or default_metric_exclusion_pattern,
-                default=False,
-            )
-
-            tag_name = config.get('tag', default_component_tag)
-            if not isinstance(tag_name, str):
-                raise ConfigurationError('The `tag` setting must be a string')
-
-            component_data[component] = (
-                tag_name,
-                lambda metric, include_metric=include_metric, exclude_metric=exclude_metric: include_metric(metric)
-                and not exclude_metric(metric),
-            )
-
-        self._components = component_data
-
-    @staticmethod
-    def compile_metric_patterns(config, field):
-        metric_patterns = config.get(field, [])
-        if not isinstance(metric_patterns, list):
-            raise ConfigurationError('The `{}` setting must be an array'.format(field))
-
-        patterns = []
-        for i, metric_pattern in enumerate(metric_patterns, 1):
-            if not isinstance(metric_pattern, str):
-                raise ConfigurationError('Pattern #{} in `{}` setting must be a string'.format(i, field))
-
-            # Ensure dots are treated as literal
-            metric_pattern = metric_pattern.replace('\\.', '.').replace('.', '\\.')
-
-            # Get rid of any explicit start modifiers
-            metric_pattern = metric_pattern.lstrip('^')
-
-            # We only search on `<CATEGORY>.<KEY>`
-            metric_pattern = re.sub(r'^sonarqube(\\.)?', '', metric_pattern)
-            if not metric_pattern:
-                raise ConfigurationError('Pattern #{} in `{}` setting must be more specific'.format(i, field))
-
-            # Match from the start by default
-            metric_pattern = '^{}'.format(metric_pattern)
-
-            patterns.append(metric_pattern)
-
-        return re.compile('|'.join(patterns)) if patterns else None
-
-    @staticmethod
-    def create_metric_matcher(pattern, default):
-        if pattern is None:
-
-            def metric_matcher(metric):
-                return default
-
-        else:
-
-            def metric_matcher(metric):
-                return not not pattern.search(metric)
-
-        return metric_matcher
-
-    @staticmethod
-    def is_valid_metric(metric):
-        return (
-            not metric['hidden']
-            and metric['type'] in NUMERIC_TYPES
-            # https://github.com/DataDog/integrations-core/pull/8552
-            and not metric['key'].startswith('new_')
+    def _process_project(self, project_key, project_config, all_metrics):
+        self.log.debug(
+            'processing matched project \'%s\' with config \'%s\'',
+            project_key,
+            project_config,
+        )
+        metrics_discovery_filter = DiscoveryFilter(
```

Comment:
This seems unnecessarily complex for users to configure. Why not make `discovery` a boolean and rely on existing patterns?

Reply:
It's not as simple as that, I'm afraid.

Reply:
It seems entirely possible; what am I missing?